File: Language/CSharp/Behavioral/Memento/Example3/Memento/Memento/Memento.cs
using System;

namespace Memento
{
    public class Memento : IMemento
    {
        private string _state;
        private DateTime _date;
        
        public Memento(string state)
        {
            this._state = state;
            this._date = DateTime.Now;
        }

        public string GetName()
        {
            return $"{this._date} / ({this._state.Substring(0, 9)})...";
        }

        public string GetState()
        {
            return this._state;
        }

        public DateTime GetDate()
        {
            return this._date;
        }
    }
}

File: TheRayCode/PHP/Behavioral/Memento/Example/Memento.php
<?php

namespace TheRayCode\DesignPatterns\Memento;

/**
 * The Memento interface provides a way to retrieve the memento's metadata, such
 * as creation date or name. However, it doesn't expose the Originator's state.
 */


interface Memento
{
    public function getName(): string;

    public function getDate(): string;
}